test/Spec.hs

```diff
@@ -37,43 +37,70 @@ import qualified Semantic.Stat.Spec
 import Semantic.Config (defaultOptions, optionsLogLevel)
 import Semantic.Task (withOptions, TaskSession(..))
 import Test.Hspec
+import Test.Tasty as Tasty
+import Test.Tasty.Hspec as Tasty
+import Test.Tasty.Options as Tasty
+
+tests :: TaskSession -> [TestTree]
+tests session =
+  [ Integration.Spec.spec session
+  , Semantic.CLI.Spec.spec
+  ]
+
+-- We can't bring this out of the IO monad until we divest
+-- from hspec, since testSpec operates in IO.
+allTests :: TaskSession -> IO TestTree
+allTests session = do
+  let nativeSpecs = tests session
+  asTastySpecs <- Tasty.testSpecs $ legacySpecs session
+  let allSpecs = nativeSpecs <> asTastySpecs
+  pure . Tasty.localOption Tasty.Success $ testGroup "semantic" allSpecs
+
+-- If you're writing new test modules, please don't add to this
+-- stanza: it is only there to prevent massive rewrites, and is
+-- converted into a Tasty TestTree in 'main'. (Quoth the tasty-hspec
+-- documentation: "hspec and tasty serve similar purposes; consider
+-- using one or the other.") Instead,
```

Comment:
1. Thank you for these docs.
2. Final line seems to be cut off.

Reply:
Oops